File: src/main/java/de/jepfa/obfusser/ui/common/NotificationHelper.java
package de.jepfa.obfusser.ui.common;

import android.app.Activity;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import android.os.Build;

public class NotificationHelper {

    public static final String CHANNEL_ID_BACKUP = "de.jepfa.notificationchannel.backup";
    public static final int NOTIFICATION_ID_BACKUP_SUCCESS = 1001;

    public static void createNotificationChannel(Context context, String channelId, String name) {
        // Create the NotificationChannel, but only on API 26+ because
        // the NotificationChannel class is new and not in the support library
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            int importance = NotificationManager.IMPORTANCE_DEFAULT;
            NotificationChannel channel = new NotificationChannel(channelId, name, importance);
            NotificationManager notificationManager = context.getSystemService(NotificationManager.class);
            notificationManager.createNotificationChannel(channel);
        }
    }
}

File: src/main/java/de/jepfa/obfusser/ui/common/NotificationHelper.kt
package de.jepfa.obfusser.ui.common

import android.app.Activity
import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.os.Build

object NotificationHelper {

    val CHANNEL_ID_BACKUP = "de.jepfa.notificationchannel.backup"
    val NOTIFICATION_ID_BACKUP_SUCCESS = 1001

    fun createNotificationChannel(context: Context, channelId: String, name: String) {
        // Create the NotificationChannel, but only on API 26+ because
        // the NotificationChannel class is new and not in the support library
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val importance = NotificationManager.IMPORTANCE_DEFAULT
            val channel = NotificationChannel(channelId, name, importance)
            val notificationManager = context.getSystemService(NotificationManager::class.java)
            notificationManager!!.createNotificationChannel(channel)
        }
    }
}
